CurrencyConverterService/RatesRecipient.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Net;
+using CurrencyConverterService.Models;
+using System.IO;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace CurrencyConverterService.Models
+{
+    class RatesRecipient
+    {
+        Context dataBase;
+        Logging logging;
+        List<Rates> rates;
+        CurrencysRecipient currencys;
+        IEnumerable<Currency> currencysDB;
+        Dictionary<int, int> currencyNumbers;
+        List<string> Name;
+        public RatesRecipient()
+        {
+            this.dataBase = new Context();
+            this.logging = new Logging();
+            this.rates = new List<Rates>();
+            this.currencys = new CurrencysRecipient();
+            this.currencyNumbers = new Dictionary<int, int>();
+            this.Name = new List<string>();
+        }
+
+        public List<Rates> GetRatesFromNBRB()
+        {
+            foreach (var currencyNumber in CurrencyNumbers())
+            {
+                if (LastDate(currencyNumber.Key).ToString("yyyy-M-d") != DateTime.Today.AddDays(+1).ToString("yyyy-M-d"))
+                {
+                    try
+                    {
+                        var Http = $"http://www.nbrb.by/API/ExRates/Rates/Dynamics/" +
+                        $"{currencyNumber.Value}?startDate={LastDate(currencyNumber.Key).ToString("yyyy-M-d")}&endDate={DateTime.Now.ToString("yyyy-M-d")}";
+                        var request = WebRequest.Create(Http);
+                        request.Method = "GET";
+                        HttpWebResponse responseObjectGet = null;
+                        responseObjectGet = (HttpWebResponse)request.GetResponse();
+                        string json = null;
+                        using (var stream = responseObjectGet.GetResponseStream())
+                        {
+                            var streamReader = new StreamReader(stream);
+                            json = streamReader.ReadToEnd();
+                            streamReader.Close();
+                        }
+                        foreach (var rate in JsonConvert.DeserializeObject<List<Rates>>(json))
+                        {
+                            rate.Bit = rate.Ask;
+                            rate.CurrencyId = currencyNumber.Key;
+                            rates.Add(rate);
+                        }
+                    }
+                    catch (Exception exception)
+                    {
+                        logging.AddError(exception.ToString());
+                    }
+                    
+                }
+                else if (LastDate(currencyNumber.Key).ToString("yyyy-M-d") == DateTime.Now.AddDays(+1).ToString("yyyy-M-d"))
```

Comment:
this else construction looks weird. I don't ever try to figure out what it is should do